### ISSUE TITLE
Components.Create<T>() creates components on children as well.

### ISSUE THREAD

**Issue by ihavefoxdie** [Facepunch/sbox-issues#5230]
### Describe the bug

When Components.Create<T>() is used on a GameObject, it creates the specified component on every child GameObject, too.
The parent GameObject is spawned with the NetworkSpawn method. Also, in my case, the creation is broadcasted.

### To Reproduce

1. Create a GameObject.
2. Create a child GameObject within it.
3. Try to create any component (I tried CameraComponent for testing as well) via the Components.Create method on the parent.

example
![image](https://github.com/Facepunch/sbox-issues/assets/59480394/fe9f5085-61ac-4813-8bdc-6212e8f8082f)

### Expected behavior

It should create specified component only on the GameObject we use it on.

### Media/Files

![image](https://github.com/Facepunch/sbox-issues/assets/59480394/911c41cb-ba30-4d41-8f6d-b25487f04961)

**Comment by kurozael** [comment]
Curious but does it _actually_ create the component on each of the children, or is it perhaps a display issue for the Inspector? Have you tried doing it but adding a simple component that prints out its GameObject's id?

**Comment by ihavefoxdie** [comment]
> Curious but does it _actually_ create the component on each of the children, or is it perhaps a display issue for the Inspector? Have you tried doing it but adding a simple component that prints out its GameObject's id?

Just did that.
Component code:
```
using Sandbox;
namespace HideAndSeek;

public class TestingBug : Component
{
	protected override void OnFixedUpdate()
	{
		base.OnFixedUpdate();
		Log.Info( this.GameObject.Id );
	}
}
```
Result:
![image](https://github.com/Facepunch/sbox-issues/assets/59480394/c722ec10-5b4e-49c9-aaea-a122516d71dc)

**Comment by ihavefoxdie** [comment]
I think me having this issue might actually be due to children having the same tags as parents. I tested this on an empty project and failed to reproduce the issue.

I used ```Scene.GetAllObjects( false ).Where( x => x.Tags.Has( "seekers" ) ); ``` and it returned a collection of every object with the tag specified, including object's children. Is it ok that child GameObjects fully inherit parent's tags, even though I haven't tagged any child specifically? I think this used not to be the case. Sorry If I am wrong.
